lib/CL/pocl_cl.h

```diff
@@ -413,6 +413,7 @@ struct _cl_device_id {
      we need to generate work-item loops to execute all the work-items
      in the WG, otherwise the hardware spawns the WIs. */
   cl_bool spmd;
+  cl_bool workgroup_pass;
```

Comment:
Please add a comment to this one. The workgroup pass serves multiple purposes at the moment. It generates the launcher functions and also converts the work item id global variable placeholders to context struct accesses.